### PR TITLE
Fix #8822 - new warning to explain why Standard62.1Summary may not be available in Output:Table:SummaryReports

### LINKED ISSUE #8822: Standard62.1Summary not be treated as valid report name in Output:Table:SummaryReports
Issue overview
--------------
Added the summary report key Standard62.1Summary into Output:Table:SummaryReports object and run the simulation for a model, EnergyPlus has reported a severe error:

   ** Severe  ** Output:Table:SummaryReports Field[6]="STANDARD62.1SUMMARY", invalid report name -- will not be reported.

The simulation still runs but there is not a Standard 62.1 Summary generated.

### Details
Some additional details for this issue (if relevant):
 - Platform: Windows 7 Professional
 - Version of EnergyPlus: 9.4.0

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

- Fix #8822 - new warning to explain why Standard62.1Summary may not be available in Output:Table:SummaryReports

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -1694,10 +1694,17 @@ void GetInputOutputTableSummaryReports(EnergyPlusData &state)
...
+                    ShowWarningError(state,
+                        format("{} Field[{}]=\"Standard62.1Summary\", Report is not enabled.", CurrentModuleObject, iReport));
+                    ShowContinueError(state, "Do Zone Sizing or Do System Sizing must be enabled in SimulationControl.");
+
+                } else {
+                    ShowSevereError(
+                        state,
+                        format("{} Field[{}]=\"{}\", invalid report name -- will not be reported.", CurrentModuleObject, iReport, AlphArray(iReport)));
+                    //      ErrorsFound=.TRUE.
+                }
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabular.cc]: New warning to explain.

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 2]
```diff
@@ -9776,3 +9779,69 @@ TEST_F(SQLiteFixture, OutputReportTabularTest_EscapeHTML)
...
+        " Standard62.1Summary; !- Report 1 Name",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    // These are already set to these values, but be explicit.
+    // Because DoZoneSizing and DoSystemSizing are both false Standard62.1Summary is **not** enabled
+    state->files.outputControl.tabular = true;
+    state->dataGlobal->DoZoneSizing = false;
+    state->dataGlobal->DoSystemSizing = false;
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:9896]: Test that the warning **is** issued when sizing calcs are **NOT** requested

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 3]
```diff
@@ -9776,3 +9779,69 @@ TEST_F(SQLiteFixture, OutputReportTabularTest_EscapeHTML)
...
+        "Output:Table:SummaryReports,",
+        " Standard62.1Summary; !- Report 1 Name",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    // Because DoZoneSizing is true Standard62.1Summary **is** enabled
+    state->files.outputControl.tabular = true;
+    state->dataGlobal->DoZoneSizing = true;
+    state->dataGlobal->DoSystemSizing = false;
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:9930]: Test that the warning is **NOT** issued when sizing calcs **are** requested

File: src/EnergyPlus/OutputReportTabular.cc (hunk 1 above)
Comment (jmarrec) [reply src/EnergyPlus/OutputReportTabular.cc]: ref code : https://github.com/NREL/EnergyPlus/blob/e936516a4cbf93937c945019c80e048dcc77d6a9/src/EnergyPlus/OutputReportPredefined.cc#L935-L937

File: doc/input-output-reference/src/standard-output-reports/output-table-summaryreports.tex [hunk 4]
```diff
@@ -397,7 +397,11 @@ \subsubsection{Predefined Annual Summary Reports}\label{predefined-annual-summar
 
 \paragraph{Standard 62.1 Summary}\label{standard-62.1-summary}
 
-The Standard 62.1 Summary (key: Standard62.1Summary) produces a report that is consistent with many of the outputs needed when doing calculations consistent with ASHRAE Standard 62.1-2010. The report is generated when sizing calculations are specified. The abbreviations used in the report are consistent with the abbreviations used in Appendix A4 of the Standard. The following tables are part of the report:
+The Standard 62.1 Summary (key: Standard62.1Summary) produces a report that is consistent with many of the outputs needed when doing calculations consistent with ASHRAE Standard 62.1-2010.
+
+\textbf{Note: The report is only generated when sizing calculations are specified.}: in \hyperref[simulationcontrol]{SimulationControl}, ``Do Zone Sizing'' or ``Do System Sizing'' must be set to ``Yes''.
+
+The abbreviations used in the report are consistent with the abbreviations used in Appendix A4 of the Standard. The following tables are part of the report:
```
Comment (jmarrec) [inline doc/input-output-reference/src/standard-output-reports/output-table-summaryreports.tex:404]: image

File: doc/output-details-and-examples/src/output-files/eplustbl-lt-ext-gt.tex [hunk 5]
```diff
@@ -2751,7 +2751,11 @@ \subsection{Sensible Heat Gain Summary}\label{sensible-heat-gain-summary}
...
 
-This report is more fully described in the Input Output Reference. The Standard 62.1 Summary produces a report that is consistent with many of the outputs needed when doing calculations consistent with ASHRAE Standard 62.1-2010. The report is generated when sizing calculations are specified. The abbreviations used in the report are consistent with the abbreviations used in Appendix A4 of the Standard. Directly following is an example of the report. The key used to obtain this report is Standard62.1Summary.
+This report is more fully described in the Input Output Reference. The Standard 62.1 Summary produces a report that is consistent with many of the outputs needed when doing calculations consistent with ASHRAE Standard 62.1-2010.
+
+The abbreviations used in the report are consistent with the abbreviations used in Appendix A4 of the Standard. Directly following is an example of the report. The key used to obtain this report is Standard62.1Summary.
+
+\textbf{Note: The report is only generated when sizing calculations are specified.}: in \hyperref[simulationcontrol]{SimulationControl}, ``Do Zone Sizing'' or ``Do System Sizing'' must be set to ``Yes''.
```
Comment (jmarrec) [inline doc/output-details-and-examples/src/output-files/eplustbl-lt-ext-gt.tex:2758]: image

**Comment by nrel-bot-2c** [comment]
@jmarrec @lgentile it has been 28 days since this pull request was last updated.

**Comment by JasonGlazer** [comment]
@jmarrec I was going to start reviewing this but noticed that it had some formatting checks failing. Can you fix those and the merge conflict? Thanks.

**Comment by jmarrec** [comment]
@JasonGlazer Merged develop and clang-formatted.

**Comment by JasonGlazer** [comment]
Thanks @jmarrec

**Review by JasonGlazer — APPROVED** [review]
Looks good @jmarrec

**Comment by JasonGlazer** [comment]
This is ready to be merged.

**Comment by jmarrec** [comment]
Thanks for the review @JasonGlazer !

**Review by mjwitte — APPROVED** [review]

**Pull request merged**